### DISCUSSION TITLE
Which architecture for a 50-120px objects on a 4096x4096 images ?

### DISCUSSION THREAD

**Discussion by ExtReMLapin** [ultralytics/yolov5#9862] [Q&A] [answered]
Hello,

I have fairly big images, with an original size of around 6000x4000 pixels, i used to be doing tiling on it, cutting images of 1024x1024 and with a moving window (overlap fo 128px) moving around the image to train my model. 

I have meh results with this but it's still better than the "feed the whole image at 4096px and try to detect 50-120px wide items".

I already gave a try with r34, p2 with each time multiple different times, I had to scale down p2 model close to nano model scale settings, yet the tiling options always get a better result.

![image](https://user-images.githubusercontent.com/3909752/196750720-8ca57b2d-ca32-4710-9e0f-e4c6e79e9432.png)
![image](https://user-images.githubusercontent.com/3909752/196751827-54bef8fb-65aa-4ad3-8b35-f53a90ae343a.png)

Attached, an example of annotation file, original image (I cannot share) has the following resolution : 4288x2848
[18-040.txt](https://github.com/ultralytics/yolov5/files/9822462/18-040.txt)

![labels](https://user-images.githubusercontent.com/3909752/196785343-734d58d6-69a7-435d-87f3-f325e04cf70c.jpg)
![labels_correlogram](https://user-images.githubusercontent.com/3909752/196785345-ba93cd4d-1bf4-4bef-bd7a-61cca279780d.jpg)

**Comment by glenn-jocher** [comment, marked as answer]
@ExtReMLapin train and detect should be done at similar object sizes, so if you want to detect at 4096 you should also train at that (possibly at batch-size 4 or 2 or 1 etc.)

**Reply by ExtReMLapin** [reply]
The metrics here are sent from yolov5 to wandb.
Unless I'm missing something, it's pictures evaluated from the `val` set after each epoch, and using `--img 4096` both resize train, test and val test so there should no be any issue with that. As it trains on 4096px, test on 4096px and val on 4096px (4096 or whatever is in the run name in the metrics above)

My point was I know there are layers that works better with some objets size, is there any model that can fits my needs ?

**Reply by glenn-jocher** [reply]
Oh ok got it! Then yes if you already training at 4096 everything should be fine. 

For small objects you can try P2 models, but these are intended for very small objects, the layer has stride 4:
https://github.com/ultralytics/yolov5/blob/master/models/hub/yolov5-p2.yaml

**Reply by ExtReMLapin** [reply]
Alright, thank you !
